### PR TITLE
Fix gen proposal reply

### PR DESCRIPTION
**Pull request by wwongkamjan**
as Denis raised
```
Traceback (most recent call last):
 File "/model/src/model_server/baseline_bots/run_bot.py", line 197, in <module>
  asyncio.run(launch(hostname=host, port=port, game_id=game_id,power_name=power, bot_type=bot_type, outdir=outdir))
 File "/data/env3.7/lib/python3.7/asyncio/runners.py", line 43, in run
  return loop.run_until_complete(main)
 File "/data/env3.7/lib/python3.7/asyncio/base_events.py", line 573, in run_until_complete
  return future.result()
 File "/model/src/model_server/baseline_bots/run_bot.py", line 80, in launch
  await play(hostname, port, game_id, power_name, bot_type, outdir)
 File "/model/src/model_server/baseline_bots/run_bot.py", line 139, in play
  ret_data = await bot(rcvd_messages)
 File "/data/env3.7/lib/python3.7/site-packages/tornado/gen.py", line 782, in run
  yielded = self.gen.send(value)
 File "/model/src/model_server/baseline_bots/src/baseline_bots/bots/smart_order_accepter_bot.py", line 554, in __call__
  best_proposer, valid_proposal_orders, msgs_data
 File "/model/src/model_server/baseline_bots/src/baseline_bots/bots/smart_order_accepter_bot.py", line 106, in gen_proposal_reply
  and self.stance.get_stance()[self.power_name][proposer] >= 0
 File "/data/env3.7/lib/python3.7/site-packages/stance_vector/stance_extraction.py", line 311, in get_stance
  self.territories = self.extract_terr()
 File "/data/env3.7/lib/python3.7/site-packages/stance_vector/stance_extraction.py", line 44, in extract_terr
  terr = {n: unit2loc(self.game.get_phase_history()[-1].state['units'][n]) for n in self.nations}
 File "/data/env3.7/lib/python3.7/site-packages/stance_vector/stance_extraction.py", line 44, in <dictcomp>
  terr = {n: unit2loc(self.game.get_phase_history()[-1].state['units'][n]) for n in self.nations}
TypeError: '_asyncio.Future' object is not subscriptable
```

I also fix some bugs in gameplay_framework, it has a infinite loop there

### CODE REVIEW

File: containers/allan_dip_bot/Dockerfile [hunk 1]
```diff
@@ -21,7 +21,7 @@ RUN chmod -R 777 /model/src/model_server/bot_neurips2019-sl_model
 # Clone repos
 RUN git clone https://github.com/SHADE-AI/diplomacy.git
 RUN git clone https://github.com/SHADE-AI/research.git
-RUN git clone https://github.com/ALLAN-DIP/baseline_bots.git
+RUN git clone --branch fix_gen_proposal_reply https://github.com/ALLAN-DIP/baseline_bots.git
```
Comment (trigaten) [inline containers/allan_dip_bot/Dockerfile:24]: A specific branch should not be getting cloned here

File: requirements.txt [hunk 2]
```diff
@@ -5,5 +5,5 @@ git+https://github.com/trigaten/DAIDE.git
 karma_sphinx_theme
 black
 isort
-git+https://github.com/ALLAN-DIP/stance_vector.git
+git+https://github.com/ALLAN-DIP/stance_vector.git@test_stance
```
Comment (trigaten) [inline requirements.txt:8]: why is a test branch being used in requirements?

File: containers/allan_dip_bot/Dockerfile (hunk 1 above)
Comment (wwongkamjan) [reply containers/allan_dip_bot/Dockerfile:24]: I will remove this later. This is just to build docker and test the bot online

File: requirements.txt (hunk 2 above)
Comment (wwongkamjan) [reply requirements.txt:8]: Similarly, I will remove this later.

**Review by kartik2112 — APPROVED** [review]

**Pull request merged**